Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.cpp

```diff
@@ -209,21 +209,21 @@ bool SourceBufferPrivateGStreamer::precheckInitializationSegment(const Initializ
         GRefPtr<GstCaps> initialCaps = videoTrackInfo->initialCaps();
         ASSERT(initialCaps);
         if (!m_tracks.contains(trackInfo.track->id()))
-            m_tracks.try_emplace(trackInfo.track->id(), MediaSourceTrackGStreamer::create(TrackPrivateBaseGStreamer::TrackType::Video, trackInfo.track->id(), videoTrackInfo->stringId(), WTFMove(initialCaps)));
+            m_tracks.add(trackInfo.track->id(), MediaSourceTrackGStreamer::create(TrackPrivateBaseGStreamer::TrackType::Video, trackInfo.track->id(), videoTrackInfo->stringId(), WTFMove(initialCaps)));
```

Comment:
Here `HashMap::ensure()` can be used maybe?

Reply:
Originally the code was using `m_tracks.add`.

https://github.com/WebKit/WebKit/pull/20901/files#diff-cbb52de8c051ac0defaf4bb8ed562696b3a14327efdcb067be7b23e912c432d9L219

Nevertheless, I tried using `m_tracks.ensure` but I got a build error:

```bash
WTF/Headers/wtf/HashMap.h:246:65: error: type 'WTF::Ref<WebCore::MediaSourceTrackGStreamer>' does not provide a call operator
        ValueTraits::ValueTraits::assignToEmpty(location.value, functor());
                                                                ^~~~~~~
WTF/Headers/wtf/HashTable.h:938:25: note: in instantiation of function template specialization 'WTF::HashMapEnsureTranslator<WTF::HashMap<unsigned long, WTF::RefPtr<WebCore::MediaSourceTrackGStreamer>>::KeyValuePairTraits, WTF::DefaultHash<unsigned long>>::translate<WTF::Key
        HashTranslator::translate(*entry, std::forward<T>(key), std::forward<Extra>(extra));
                        ^
WTF/Headers/wtf/HashMap.h:402:28: note: in instantiation of function template specialization 'WTF::HashTable<unsigned long, WTF::KeyValuePair<unsigned long, WTF::RefPtr<WebCore::MediaSourceTrackGStreamer>>, WTF::KeyValuePairKeyExtractor<WTF::KeyValuePair<unsigned long, WTF::
    return m_impl.template add<HashMapEnsureTranslator<KeyValuePairTraits, HashFunctions>>(std::forward<K>(key), std::forward<F>(functor));
                           ^
WTF/Headers/wtf/HashMap.h:465:12: note: in instantiation of function template specialization 'WTF::HashMap<unsigned long, WTF::RefPtr<WebCore::MediaSourceTrackGStreamer>>::inlineEnsure<unsigned long, WTF::Ref<WebCore::MediaSourceTrackGStreamer>>' requested here
    return inlineEnsure(std::forward<KeyType>(key), std::forward<Functor>(functor));
           ^
../../../Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.cpp:225:22: note: in instantiation of function template specialization 'WTF::HashMap<unsigned long, WTF::RefPtr<WebCore::MediaSourceTrackGStreamer>>::ensure<WTF::Ref<WebCore::MediaSourceTrac
            m_tracks.ensure(trackInfo.track->id(), MediaSourceTrackGStreamer::create(TrackPrivateBaseGStreamer::TrackType::Video, trackInfo.track->id(), videoTrackInfo->stringId(), WTFMove(initialCaps)));
                     ^
1 error generated.
```

Reply:
Well, the second argument has to be a lambda...

Anyways, whatever. r+